internal/provider/resource/cert_manager_ca_certificate.go

```diff
@@ -0,0 +1,328 @@
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *infisical.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *certManagerCACertificateResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	if !r.client.Config.IsMachineIdentityAuth {
+		resp.Diagnostics.AddError(
+			"Unable to create CA certificate",
+			"Only Machine Identity authentication is supported for this operation",
+		)
+		return
+	}
+
+	var plan certManagerCACertificateResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	notBefore, err := time.Parse(time.RFC3339, plan.NotBefore.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid not_before format", "not_before must be in RFC3339 format (e.g., '2024-01-01T00:00:00Z')")
+		return
+	}
+
+	notAfter, err := time.Parse(time.RFC3339, plan.NotAfter.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid not_after format", "not_after must be in RFC3339 format (e.g., '2034-01-01T00:00:00Z')")
+		return
+	}
+
+	if notAfter.Before(notBefore) {
+		resp.Diagnostics.AddError("Invalid date range", "not_after must be after not_before")
+		return
+	}
+
+	generateRequest := infisical.GenerateCACertificateRequest{
+		CaId:      plan.CaId.ValueString(),
+		NotBefore: plan.NotBefore.ValueString(),
+		NotAfter:  plan.NotAfter.ValueString(),
+	}
+
+	if !plan.MaxPathLength.IsNull() {
+		maxPathLength := int(plan.MaxPathLength.ValueInt64())
+		generateRequest.MaxPathLength = &maxPathLength
+	}
+
+	if !plan.ParentCaId.IsNull() && plan.ParentCaId.ValueString() != "" {
+		generateRequest.ParentCaId = plan.ParentCaId.ValueString()
+	}
+
+	certificate, err := r.client.GenerateCACertificate(generateRequest)
+	if err != nil {
+		resp.Diagnostics.AddError("Error generating CA certificate", err.Error())
+		return
+	}
+
+	plan.Id = types.StringValue(fmt.Sprintf("%s:%s", plan.CaId.ValueString(), certificate.SerialNumber))
+	plan.Certificate = types.StringValue(certificate.Certificate)
+	plan.CertificateChain = types.StringValue(certificate.CertificateChain)
+	plan.SerialNumber = types.StringValue(certificate.SerialNumber)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *certManagerCACertificateResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	if !r.client.Config.IsMachineIdentityAuth {
+		resp.Diagnostics.AddError(
+			"Unable to read CA certificate",
+			"Only Machine Identity authentication is supported for this operation",
+		)
+		return
+	}
+
+	var state certManagerCACertificateResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	certificate, err := r.client.GetCACertificate(infisical.GetCACertificateRequest{
+		CaId: state.CaId.ValueString(),
+	})
```

Comment:
wait.. why do we query it just by the CA ID? A CA can have more than one certificate right?

Reply:
Ah looks like this is indeed causing a bug when I have more than one CA certificate 
<img width="737" height="640" alt="image" src="https://github.com/user-attachments/assets/ab8670ca-422d-4157-92ab-3c79c63fd7fa" />